extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -789,29 +789,36 @@ export class AmpConsent extends AMP.BaseElement {
     if (!this.isGranularConsentExperimentOn_) {
       return Promise.resolve(true);
     }
-    return this.getPurposeConsentRequired_().then((purposeConsentRequired) => {
-      // True if there are no required purposes
-      if (!purposeConsentRequired || !purposeConsentRequired.length) {
-        return true;
-      }
-      const storedPurposeConsents = consentInfo['purposeConsents'];
-      // False if there are no stored purposes
-      if (
-        !storedPurposeConsents ||
-        Object.keys(storedPurposeConsents).length <
-          purposeConsentRequired.length
-      ) {
-        return false;
-      }
-      // Check if we have a stored consent for each purpose required
-      for (let i = 0; i < purposeConsentRequired.length; i++) {
-        const purpose = purposeConsentRequired[i];
-        if (!hasOwn(storedPurposeConsents, purpose)) {
+    return this.getPurposeConsentRequired_()
+      .then((purposeConsentRequired) => {
+        // True if there are no required purposes
+        if (!purposeConsentRequired || !purposeConsentRequired.length) {
+          return true;
+        }
+        const storedPurposeConsents = consentInfo['purposeConsents'];
+        // False if there are no stored purposes
+        if (
+          !storedPurposeConsents ||
+          Object.keys(storedPurposeConsents).length <
+            purposeConsentRequired.length
+        ) {
           return false;
         }
-      }
-      return true;
-    });
+        // Check if we have a stored consent for each purpose required
+        for (let i = 0; i < purposeConsentRequired.length; i++) {
+          const purpose = purposeConsentRequired[i];
+          if (!hasOwn(storedPurposeConsents, purpose)) {
+            return false;
+          }
+        }
+        return true;
+      })
+      .then((hasPurposeConsents) => {
```

Comment:
As far as I can tell, nothing in the above `.then` handler is asynchronous or returning a promise. Rather than extending the promise chain, this chunk could live in the function/handler above

Reply:
Just thought it would be cleaning to separate these ideas, but I can fix and also insure clarity.